jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -54,6 +56,18 @@ def __init__(self, *args, **kwargs):
                                           'subset_items',
                                           'subset_selected',
                                           default_text="Create New")
+        self.all_subsets_with_subregions = {}
+
+    @property
+    def user_api(self):
+        return PluginUserApi(self, expose=('subset_select', 'subset_definitions',
+                                           'get_all_subsets_with_subregions'))
```

Comment:
maybe we should defer making this public for now since we're expecting significant development in the plugin going forward?  If so, you can just get rid of this property and change any internal calls to go through `plugin._obj` instead.

Reply:
Which development did you have in mind? Moving the recursion from here to `utils` shouldn't change any of the behavior in these exposed values.

Reply:
> Which development did you have in mind?

Nested subsets support, ability to create new subsets from the plugin, etc.  But if we want to commit to the API for the three methods/attributes, then that is fine with me... but then this PR should probably also update the docstrings for the class itself and those methods.